source/libspirv.cpp

```diff
@@ -111,8 +112,15 @@ bool SpirvTools::Validate(const uint32_t* binary,
 bool SpirvTools::Validate(const uint32_t* binary, const size_t binary_size,
                           const ValidatorOptions& options) const {
   spv_const_binary_t the_binary{binary, binary_size};
-  return spvValidateWithOptions(impl_->context, options, &the_binary,
-                                nullptr) == SPV_SUCCESS;
+  spv_diagnostic diagnostic = nullptr;
+  bool valid = spvValidateWithOptions(impl_->context, options, &the_binary,
+                                      &diagnostic) == SPV_SUCCESS;
+  if (!valid && impl_->context->consumer) {
+    impl_->context->consumer.operator()(
```

Comment:
So, this is just to feed messages back from the c api to the consumer? How did this work before?

Reply:
See my other comment in tools/opt/opt.cpp.